ui_testing/testcases/basic_tests/test04_testunits.py

```diff
@@ -1017,36 +919,29 @@ def test025_edit_approach_overview_button(self):
         self.assertEqual(self.base_selenium.get_url(), '{}testUnits'.format(self.base_selenium.url))
         self.base_selenium.LOGGER.info('clicking on Overview confirmed')
 
-    @parameterized.expand(['Quantitative', 'Qualitative', 'MiBi'])
-    def test_028_changing_testunit_type_update_fields_accordingly(self, testunit_type):
+    def test_027_changing_testunit_type_update_fields_accordingly(self):
         """
-        New: Test unit: Type Approach: When I change type from edit mode, the values should changed according to this type that selected 
-        When I change type from edit mode, the values should changed according to this type that selected 
-        LIMS-3680
+        New: Test unit: Type Approach: When I change type from edit mode, the values
+        should changed according to this type that selected
 
-        comment: this case will be handled in create
+        LIMS-3680
         """
-
-        if testunit_type == 'MiBi':
-            testunit_type = 'Quantitative MiBi'
-
-        self.base_selenium.LOGGER.info('open testunits in create')
-        self.test_unit_page.click_create_new_testunit()
-
-        self.base_selenium.LOGGER.info('set the type to {}'.format(testunit_type))
-        self.test_unit_page.set_testunit_type(testunit_type=testunit_type)
-        self.test_unit_page.sleep_tiny()
-        self.base_selenium.LOGGER.info(
-            'set testunit type to {}, fields should be displayed as the following'.format(testunit_type))
-
-        if testunit_type == 'Quantitative':
-            self.assertTrue(self.test_unit_page.check_for_quantitative_fields())
-        elif testunit_type == 'Qualitative':
-            self.assertTrue(self.test_unit_page.check_for_qualitative_fields())
-        elif testunit_type == 'Quantitative MiBi':
-            self.assertTrue(self.test_unit_page.check_for_quantitative_mibi_fields())
-
-    def test_029_allow_user_to_change_between_specification_and_quantification(self):
+        self.info('edit random testunit')
+        self.test_unit_page.get_random_test_units()
+        self.base_selenium.get_url()
+        testunit_types = ['Quantitative', 'Qualitative', 'Quantitative MiBi']
+        for testunit_type in testunit_types:
```

Comment:
This is not correct, For loop doesn't equal the parameterized. For loop means it is only one test case, parameterized means there r n test cases.

Reply:
it's already one test case : 
https://github.com/modeso-open-source/1lims-automation/blob/LIMS-3680/ui_testing/testcases/basic_tests/test04_testunits.py#L922

Reply:
no, please separate them those are 3 different test case
as general rulse:

1. - Test case should test only one case
2. - Test case should have as minimum assertion as possible

Reply:
ok, it's a good point